src/pudl/output/ferc714.py

```diff
@@ -186,28 +186,35 @@ def filled_core_eia861__yearly_balancing_authority(
     """Modified core_eia861__yearly_balancing_authority table.
 
     This function adds rows for each balancing authority-year pair missing from the
-    cleaned core_eia861__yearly_balancing_authority table, using a dictionary of manual fixes. It
-    uses the reference year as a template. The function also removes balancing
-    authorities that are manually categorized as utilities.
+    cleaned :ref:`core_eia861__yearly_balancing_authority` table, using a dictionary
+    of manual fixes. It uses the reference year as a template. The function also removes
+    balancing authorities that are manually categorized as utilities.
     """
     df = core_eia861__yearly_balancing_authority
     index = ["balancing_authority_id_eia", "report_date"]
     dfi = df.set_index(index)
     # Prepare reference rows
-    keys = [(fix["id"], pd.Timestamp(fix["from"], 1, 1)) for fix in ASSOCIATIONS]
+    eia861_years = df["report_date"].dt.year.unique()
+    keys = [
+        (fix["id"], pd.Timestamp(fix["from"], 1, 1))
+        for fix in ASSOCIATIONS
+        if fix["from"] in eia861_years
+    ]
     refs = dfi.loc[keys].reset_index().to_dict("records")
     # Build table of new rows
     # Insert row for each target balancing authority-year pair
     # missing from the original table, using the reference year as a template.
     rows: list[dict[str, Any]] = []
-    for ref, fix in zip(refs, ASSOCIATIONS, strict=True):
+    for ref, fix in zip(
+        refs, [fx for fx in ASSOCIATIONS if fx["from"] in eia861_years], strict=True
+    ):
         for year in range(fix["to"][0], fix["to"][1] + 1):
             key = (fix["id"], pd.Timestamp(year, 1, 1))
```

Comment:
would it make sense to skip iterations here when year is not in eia861_years, instead of filtering at 216?